common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java

```diff
@@ -57,4 +60,28 @@ public static String urlFormDecode(final String source) throws UnsupportedEncodi
 
         return URLDecoder.decode(source, ENCODING_UTF8);
     }
+
+    /**
+     * Counts the number of occurrences of one String in another.
+     *
+     * @param str
+     * @param subString
+     * @return int
+     */
+    public static int countMatches(@NonNull final String str, @Nullable final String subString) {
+        int count = 0;
+
+        if (isNullOrEmpty(str) || isNullOrEmpty(subString)) {
+            return count;
+        }
+
+        for (int i = 0; i <= (str.length() - subString.length()); i++) {
+            if (str.substring(i, i + subString.length()).equalsIgnoreCase(subString)) {
+                count++;
+            }
+        }
+
```

Comment:
This seems to be starting at every index and then checking to see if the string is case insensitively equals to the substring starting at that index, which is maybe the least efficient way to do this thing and is also going to create a lot of Strings along the way.

We could do much, *much* better, but I think just cutting the extraneous allocations might be worthwhile - which you could do by checking the case-converted  code point at the index versus the starting case-converted code point of the substring before launching into the string-creation-and-comparison block.

Reply:
Changed this logic and add test cases. LMK if this works better.

Reply:
Is this only used in test cases?  If so, I don't care, but we should move it to a test package.

Reply:
ah yeah you're right. Will do.